File: Diagrammatize.Design.Pattern/code/cxx/Decorator/A1/StringDisplay.cpp
#include "StringDisplay.hpp"

StringDisplay::StringDisplay(const std::string& string_): string(string_) {
}

int StringDisplay::getColumns() {
    return string.length();
}

int StringDisplay::getRows() {
    return 1;
}

std::string StringDisplay::getRowText(int row) {
    if (row == 0) {
        return string;
    } else {
        return "";
    }
}

File: Diagrammatize.Design.Patterns/code/python/Decorator/A1/StringDisplay.py
from Display import Display

class StringDisplay(Display):
    def __init__(self, string):
        self.string = string

    def getColumns(self):
        return self.getStringLen(self.string)

    def getRows(self):
        return 1

    def getRowText(self, row):
        if row == 0:
            return self.string
        else:
            return None
